src/lib/crypto/kimchi_backend/common/tests/test_lookup_table_constraint_kind.ml

```diff
@@ -35,17 +50,382 @@ let test_finalize_and_get_gates_with_lookup_tables () =
+  assert (Array.length runtime_tables = 1) ;
+  let rt = runtime_tables.(0) in
+  (* Second value is chosen *)
+  assert (Tick.Field.equal rt.data.(repeated_idx) fv2) ;
+  assert (Tick.Field.equal rt.data.(other_idx) fv3)
+
+let test_compute_witness_returns_correctly_filled_runtime_tables_one_lookup () =
+  (* We have one table with ID 0, indexed from 0 to n, and we will fill with
+     some values using the constraint RuntimeLookup.
+     We start with one lookup
+  *)
+  let n = 10 in
+  let first_column = Array.init n Tick.Field.of_int in
+  let table_id = 0 in
+  let idx = Random.int n in
+  let v = Tick.Field.random () in
+  let external_values = Tick.Field.[| of_int table_id; of_int idx; v |] in
+  let cs =
+    Impl.constraint_system ~input_typ:Impl.Typ.unit ~return_typ:Impl.Typ.unit
+      (fun () () ->
+        let vtable_id =
+          Impl.exists Impl.Field.typ ~compute:(fun () -> external_values.(0))
+        in
+        let vidx =
+          Impl.exists Impl.Field.typ ~compute:(fun () -> external_values.(1))
+        in
+        let vv =
+          Impl.exists Impl.Field.typ ~compute:(fun () -> external_values.(2))
+        in
+        (* Config *)
+        add_constraint
+          (AddRuntimeTableCfg { id = Int32.of_int table_id; first_column }) ;
+        add_constraint
+          (Lookup
+             { w0 = vtable_id
+             ; w1 = vidx
+             ; w2 = vv
+             ; w3 = vidx
+             ; w4 = vv
+             ; w5 = vidx
+             ; w6 = vv
+             } ) )
+  in
+  let _ = Tick.R1CS_constraint_system.finalize cs in
+  let _witnesses, runtime_tables =
+    Tick.R1CS_constraint_system.compute_witness cs (Array.get external_values)
+  in
+  (* checking only one table has been created *)
+  assert (Array.length runtime_tables = 1) ;
+  let rt = runtime_tables.(0) in
+  (* with the correct ID *)
+  assert (Int32.(equal rt.id (of_int table_id))) ;
+  let exp_rt = Array.init n (fun i -> if i = idx then v else Tick.Field.zero) in
+  assert (Array.for_all2 Tick.Field.equal rt.data exp_rt)
+
+let test_compute_witness_returns_correctly_filled_runtime_tables_multiple_lookup
+    () =
+  (* We have one table with ID 0, indexed from 0 to n, and we will fill with
+     some values using the constraint RuntimeLookup.
+     We start with one lookup
+  *)
+  let n = 10 in
+  let first_column = Array.init n Tick.Field.of_int in
+  let table_id = 0 in
+  let exp_rt_data = Array.init n (fun _ -> Tick.Field.zero) in
+  (* nb of lookups *)
+  let m = Random.int n in
+  let external_values = Array.init (1 + (m * 2)) (fun _ -> Tick.Field.zero) in
+  let cs =
+    Impl.constraint_system ~input_typ:Impl.Typ.unit ~return_typ:Impl.Typ.unit
+      (fun () () ->
+        let vtable_id =
+          Impl.exists Impl.Field.typ ~compute:(fun () -> external_values.(0))
+        in
+        (* Config *)
+        add_constraint
+          (AddRuntimeTableCfg { id = Int32.of_int table_id; first_column }) ;
```

Comment:
very interesting

Reply:
If we can configure tables on the go, will we remove the "fixed" ones we already have and only define them when they are being used in a given circuit?

Reply:
I don't think so. [The pattern](https://github.com/o1-labs/proof-systems/blob/master/kimchi/src/circuits/lookup/lookups.rs#L414) is different, it would require other changes that we might not want now.